### PR TITLE
exchanges/order: Improve order side handling in SubmitOrder

### LINKED ISSUE #1270: Buy/Sell/Ask/Bid actions do not have an expected outcome
Instead of the requested bid, Binance performed a sell action as order.Buy is treated as Sell, But order.Bid isn't accounted for.
We need to check the issue across all exchanges and fix the actions if incorrect

### PR DESCRIPTION
**Pull request by Beadko**
# PR Description

The fix is intended to cover all order actions- Buy, Sell, Ask, Bid, Long and Short by using IsLong() instead of order.Buy. Combined the exchanges in stead of creating a PR for each one separately

Fixes #1270 

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 1]
```diff
@@ -580,11 +580,11 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (*order.S
 		return nil, err
 	}
 
-	if s.Side == order.Sell {
-		s.Side = order.Ask
+	if s.Side.IsLong() {
+		s.Side = order.Buy
```
Comment (thrasher-) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:584]:
```suggestion
		s.Side = order.Bid
```

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 2]
```diff
@@ -580,11 +580,11 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (*order.S
...
 	}
-	if s.Side == order.Buy {
-		s.Side = order.Bid
+	if s.Side.IsShort() {
+		s.Side = order.Sell
```
Comment (thrasher-) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:587]:
```suggestion
		s.Side = order.Ask
```

To match API expectation:

image

**Review by thrasher- — CHANGES_REQUESTED** [review]
Very nice! Just some minor changes

File: exchanges/btcmarkets/btcmarkets_wrapper.go (hunk 2 above)
Comment (Beadko) [reply exchanges/btcmarkets/btcmarkets_wrapper.go]: solved

**Review by thrasher- — COMMENTED** [review]
Thanks for making those changes!

Given that the other SubmitOrder PRs are related to this change, it would be good to cherry-pick their commits into this one the scope of changes is small, helps reduce PR clutter and simplifies the review process

**Comment by thrasher-** [comment]
Consolidated the other PRs into this one as per the above comment

**Review by shazbert — APPROVED** [review]
Looks good to me! Thanks!

**Pull request merged**